src/Support/Contracts/WithReferenceRule.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Nuwave\Lighthouse\Support\Contracts;
+
+interface WithReferenceRule
+{
+    public function setArgumentPath(string $argumentPath): void;
```

Comment:
This should have a docblock explaining what the argument path exactly is and how to use it.